lib/tasks/images.rake

```diff
@@ -0,0 +1,95 @@
+# frozen_string_literal: true
+
+namespace :images do
+  # Images can be lost for several reasons, for example:
+  #
+  # - file loss on disk
+  # - failed upload
+  # - file loss on bucket
+  # - file loss when changing storage
+  #
+  # If we can't recover the image, we should delete the record in the database
+  # as well. Otherwise operations like regenerating thumbnails will try to
+  # find those images every time which costs time and bandwidth.
+  task delete_orphans: :environment do
+    puts "To identify orphaned images, update the metadata first:"
+    puts ""
+    puts "  bundle exec rake paperclip:refresh:metadata CLASS=Spree::Image"
+    puts ""
+
+    orphaned_images = Spree::Image.where(attachment_file_size: 0)
+    total = Spree::Image.count
+    really_delete_images = ENV.fetch("DELETE_IMAGES", false)
+
+    if really_delete_images
+      puts "Deleting #{orphaned_images.count} out of #{total} images!"
+      [3, 2, 1].each do |i|
+        print "#{i} "
+        sleep 1
+      end
+      puts ""
+      deleted = orphaned_images.delete_all
+      puts "Done. #{deleted} image records were deleted."
+    else
+      puts "Found #{orphaned_images.count} out of #{total} images could be deleted."
+      puts "Set environment variable DELETE_IMAGES=true to delete them."
+    end
+  end
+
+  task reset_styles: :environment do
+    klass = Paperclip::Task.obtain_class
+    names = Paperclip::Task.obtain_attachments(klass)
+    styles = obtain_styles
+
+    names.each do |name|
+      Kernel.const_get(klass).attachment_definitions[name][:styles] = styles
+    end
+  end
+
+  desc "Regenerates thumbnails for a given CLASS"\
+       " (and optional ATTACHMENT and STYLES splitted by comma)."
+  task regenerate_thumbnails: :environment do
+    klass = Paperclip::Task.obtain_class
+    names = Paperclip::Task.obtain_attachments(klass)
+    styles = (ENV['STYLES'] || ENV['styles'] || '').split(',').map(&:to_sym)
```

Comment:
to regenerate styles I think we should fetch the styles from
Spree::Image.attachment_definitions[:attachment][:styles]

Reply:
more details here: https://github.com/openfoodfoundation/openfoodnetwork/pull/5729

Reply:
the detailed approach I recommend, please have a look Maikel: https://github.com/openfoodfoundation/ofn-install/pull/651#issuecomment-694985893

Reply:
This would be great but the whole reason I came up with this is that we need to regenerate thumbnails with new styles *before* we deploy the new code. If we don't do that beforehand then images won't display until they have been regenerated. That could be a whole day.

The image size is not the problem but the image format. Some instances keep the original format and have PNG images. As soon as we deploy the new standardised styles, the shop front will be looking for JPG files which don't exist yet. If you thought of that case already then I missed it somehow. This is when I wish we were in the same time zone. We could have sorted this two weeks ago...

Reply:
ok Maikel I missed that point!
Now that I understand what you are doing I think this makes perfect sense!
This PR is valid and compatible with all that I have done in my PRs :+1: so we can just go forward with the 3 PRs.